modules/git/repo_stats.go

```diff
@@ -49,86 +50,103 @@ func (repo *Repository) GetCodeActivityStats(fromTime time.Time, branch string)
 	}
 	stats.CommitCountInAllBranches = c
 
+	stdoutReader, stdoutWriter, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = stdoutReader.Close()
+		_ = stdoutWriter.Close()
+	}()
+
 	args := []string{"log", "--numstat", "--no-merges", "--pretty=format:---%n%h%n%an%n%ae%n", "--date=iso", fmt.Sprintf("--since='%s'", since)}
 	if len(branch) == 0 {
 		args = append(args, "--branches=*")
 	} else {
 		args = append(args, "--first-parent", branch)
 	}
 
-	stdout, err = NewCommand(args...).RunInDirBytes(repo.Path)
-	if err != nil {
-		return nil, err
-	}
+	err = NewCommand(args...).RunInDirTimeoutEnvFullPipelineFunc(
+		nil, -1, repo.Path,
+		stdoutWriter, nil, nil,
```

Comment:
You probably still need to store the stderr in case there is an error. Git will only send a very small message if there is an error so that's fine.